src/dmd/backend/dtype.d

```diff
@@ -0,0 +1,1754 @@
+/**
```

Comment:
Shouldn't this file replace the existing `type.d`?

Reply:
There's already a type.d.

Reply:
I know, but looks like bindings to type.c.

Reply:
It is. Unfortunately, I gave up on using .di/.d files for the compiler source because it broke all the different build systems for dmd. Right now, I'm only interested in getting everything in D in the most straightforward way practical before reorganizing it.

Reply:
Do we need .di files?

Reply:
To separate definition from implementation, yes.

Reply:
We don’t have that for the front end why do you think we need that for the back end?

Reply:
The most pragmatic answer is that's the way the C++ code was organized, and translating works a lot better if you don't try and refactor it in the process.